src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,138 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> oldNode = tail;
+        Node<T> newNode = new Node<>(oldNode,value,null);
+        tail = newNode;
+        if (oldNode != null) {
+            oldNode.next = newNode;
+        } else {
+            head = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            head.prev = new Node<>(null,value,head);
+            head = head.prev;
+            size++;
+        } else {
+            Node<T> node = getNodeByIndex(index);
+            Node<T> midValue = new Node<>(node.prev,value,node);
+            midValue.next.prev = midValue;
+            midValue.prev.next = midValue;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNodeByIndex(index);
+        T element = node.item;
+        node.item = value;
+        return element;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> currentNode = getNodeByIndex(index);
+        T removedValue = currentNode.item;
+        unlink(currentNode);
+        return removedValue;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> current = head;
+        for (int i = 0; i < size; i++) {
+            if (current.item == object || current.item != null && current.item.equals(object)) {
+                remove(i);
+                return true;
+            }
```

Comment:
you are iterating 2 times through your nodes, use `unlink()` instead of `remove()`